src/distdl/nn/pooling.py

```diff
@@ -44,13 +44,20 @@ class requires the following of the tensor partitions:
     # Pooling class for base unit of work.
     TorchPoolType = None  # noqa F821
 
-    def __init__(self, P_x, *args, **kwargs):
+    def __init__(self, P_x, buffer_manager=None, *args, **kwargs):
 
         super(DistributedPoolBase, self).__init__()
 
         # P_x is 1 x 1 x P_d-1 x ... x P_0
         self.P_x = P_x
 
+        # Back-end specific buffer manager for economic buffer allocation
+        if buffer_manager is None:
+            buffer_manager = self._distdl_backend.BufferManager()
+        elif type(buffer_manager) is not self._distdl_backend.BufferManager:
+            raise ValueError("Incompatible buffer manager.")
```

Comment:
Same as above

Reply:
Done as above, and in three more places.